content/integrations/google_cloud_bigtable/Java/_index.md

```diff
@@ -0,0 +1,377 @@
+                table.put(put);
+            }
+        }
+    }
+
+    private void deleteTable(Table table) throws Exception {
+        try (Scope ss = tracer.spanBuilder("DeleteTable").startScopedSpan()) {
+            System.out.println("Deleting the table");
+            this.admin.disableTable(table.getName());
+            this.admin.deleteTable(table.getName());
+        }
+    }
+
+    private void readRows(Table table) throws Exception {
+        try (Scope ss = tracer.spanBuilder("ReadRows").startScopedSpan()) {
+            System.out.println("Scan for all greetings:");
+            Scan scan = new Scan();
+
+            ResultScanner scanner = table.getScanner(scan);
+            for (Result row : scanner) {
+                byte[] valueBytes = row.getValue(COLUMN_FAMILY_NAME, COLUMN_NAME);
+                System.out.println('\t' + Bytes.toString(valueBytes));
+            }
+        }
+    }
+
+    private void enableOpenCensusObservability() throws IOException {
+        // Start: enable observability with OpenCensus tracing and metrics
+        // Create and register the Stackdriver Tracing exporter
+        StackdriverTraceExporter.createAndRegister(
+                StackdriverTraceConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+        // Create and register the Stackdriver Monitoring/Metrics exporter
+        StackdriverStatsExporter.createAndRegister(
+                StackdriverStatsConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+        // Register all the gRPC views
+        RpcViews.registerAllGrpcViews();
+
+        // For demo purposes, we are enabling the always sampler
+        TraceConfig traceConfig = Tracing.getTraceConfig();
+        traceConfig.updateActiveTraceParams(
+                traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample()).build());
+        // End: enable observability with OpenCensus
+    }
+}
+{{</highlight>}}
+{{<highlight xml>}}
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>io.opencensus.tutorials.bigtable</groupId>
+    <artifactId>bigtable</artifactId>
+    <packaging>jar</packaging>
+    <version>1.0-SNAPSHOT</version>
+    <name>bigtableoc</name>
+    <url>http://maven.apache.org</url>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <opencensus.version>0.15.0</opencensus.version> <!-- The OpenCensus version to use -->
+    </properties>
+
+    <dependencies>
+	<dependency>
+	    <groupId>com.google.cloud.bigtable</groupId>
+	    <artifactId>bigtable-hbase-dataflow</artifactId>
+	    <version>1.4.0</version>
```

Comment:
Instead of excluding shaded artifacts, you can add a `dependencyManagement` section to force a consistent version for OpenCensus in your maven project. For example:
```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>io.opencensus</groupId>
      <artifactId>opencensus-api</artifactId>
      <version>${opencensus.version}</version>
    </dependency>
  </dependencies>
</dependencyManagement>
```
See https://stackoverflow.com/questions/36286519/excluding-artifacts-with-maven-shade-plugin-is-not-working-as-expected.

Reply:
You need to include:
```xml
<dependency>
  <groupId>com.google.protobuf</groupId>
  <artifactId>protobuf-java</artifactId>
  <version>3.6.1</version>
</dependency>
```
in `dependencyManagement` too.

There are still netty dependency failures, I'm investigating.

Reply:
@odeke-em This is my pom.xml file: https://github.com/songy23/CloudBigtableDemo/blob/master/pom.xml. It worked fine with all the dependencies though it might be over-complicated.

Reply:
Awesome, thanks @songy23! Lots of progress when using your pom.xml file. The last problem am seeing on my computer is a netty dependency problem but we are close

On running
```shell
GOOGLE_APPLICATION_CREDENTIALS=~/Downloads/census-demos-a835580cf6cc.json mvn exec:java -Dexec.mainClass=io.opencensus.tutorials.bigtable.BigtableOpenCensus
```

I get back:
```shell
java.lang.IllegalArgumentException: SunJSSE selected, but Jetty NPN/ALPN unavailable
	at io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:223)
	at io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:189)
	at io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts.configure(GrpcSslContexts.java:171)
	at io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts.forClient(GrpcSslContexts.java:120)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder$NettyTransportFactory$DefaultNettyTransportCreationParamsFilterFactory.<init>(NettyChannelBuilder.java:558)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder$NettyTransportFactory$DefaultNettyTransportCreationParamsFilterFactory.<init>(NettyChannelBuilder.java:551)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder$NettyTransportFactory.<init>(NettyChannelBuilder.java:489)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.buildTransportFactory(NettyChannelBuilder.java:337)
	at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:405)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:206)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:157)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:149)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:151)
	at com.google.cloud.trace.v2.stub.GrpcTraceServiceStub.create(GrpcTraceServiceStub.java:70)
	at com.google.cloud.trace.v2.stub.TraceServiceStubSettings.createStub(TraceServiceStubSettings.java:99)
	at com.google.cloud.trace.v2.TraceServiceClient.<init>(TraceServiceClient.java:137)
	at com.google.cloud.trace.v2.TraceServiceClient.create(TraceServiceClient.java:118)
	at io.opencensus.exporter.trace.stackdriver.StackdriverV2ExporterHandler.createWithCredentials(StackdriverV2ExporterHandler.java:131)
	at io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter.createAndRegister(StackdriverTraceExporter.java:80)
	at io.opencensus.tutorials.bigtable.BigtableOpenCensus.enableOpenCensusObservability(BigtableOpenCensus.java:130)
	at io.opencensus.tutorials.bigtable.BigtableOpenCensus.<init>(BigtableOpenCensus.java:50)
	at io.opencensus.tutorials.bigtable.BigtableOpenCensus.main(BigtableOpenCensus.java:60)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:282)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

Reply:
Did you add netty-tcnative-boringssl-static as a runtime dependency?

Reply:
Yeah, I was using your pom.xml file and even with that update it trips out. But that's alright. Could you please help me with a screenshot of the Stackdriver traces so that we can include them in the doc? After that, I think we'll be good to merge :) Thank you for all your help.

Reply:
For example this is what the one for the Go snippet looks like
https://github.com/opencensus-integrations/opencensus-website/blob/a6d645c17182051d9eef270e4571f1c98da4c04b/static/images/cloud_bigtable_trace.png